test/commands/login.test.js

```diff
@@ -1,38 +1,103 @@
-const Nock = require('@fancy-test/nock').default;
-const { expect, test } = require('@oclif/test');
 const jwt = require('jsonwebtoken');
+const nock = require('nock');
 
-const fancy = test.register('nock', Nock);
+const testDialog = require('./test-cli');
+const LoginCommand = require('../../src/commands/login');
 
 describe('login', () => {
-  describe('with trivial mail and bad token in args', () => {
-    fancy
-      .stdout({ print: true })
-      .stderr()
-      .command(['login', '-e', 'smile@gmail.com', '-t', 'bad_token'])
-      .it('should warn about bad token', (ctx) => {
-        expect(ctx.stderr).to.contain('Invalid token. Please enter your authentication token.');
+  describe('with email in args', () => {
+    describe('with bad token in args', () => {
+      it('should display invalid token', () => testDialog({
+        command: () => LoginCommand.run(['-e', 'smile@gmail.com', '-t', 'invalid_token']),
+        dialog: [
+          { err: 'Invalid token. Please enter your authentication token.' },
+        ],
+      }));
+    });
+    describe('with valid token in args', () => {
+      const token = jwt.sign({}, 'key', { expiresIn: '1day' });
+      it('should login successful', () => testDialog({
+        command: () => LoginCommand.run(['-e', 'smile@gmail.com', '-t', token]),
+        dialog: [
+          { in: `${jwt.sign({}, 'key', { expiresIn: '1day' })}` },
+          { out: 'Login successful' },
+        ],
```

Comment:
[Thinking] By using this dialog we test two things there according to me. We test the result of the command, and what the command displays in the console. I'm not sure we should tie this two tests together. I have the feeling it will be hard to maintain / subject to bad dev experience.

Reply:
Ok. we will have the occasion to improve this.